merkletree.go

```diff
@@ -83,11 +83,17 @@ func (m *MerkleTree) Get(key string) (MerkleNode, []ProofNode) {
 
 func (m *MerkleTree) Set(key string, value []byte) {
 	index := computePrivateIndex(key)
+
+	// generate random salt per user
+	salt := make([]byte, crypto.MaxSaltSize)
+	_, _ = io.ReadFull(rand.Reader, salt)
```

Comment:
Actually, in very rare cases an error could occur if there isn't enough entropy (at least under linux): https://github.com/golang/go/blob/master/src/internal/syscall/unix/getrandom_linux.go#L27